### ISSUE TITLE
[0011] Get the `vk::SpirvType` or `vk::SpirvOpaqueType` of an expression

### ISSUE THREAD

**Issue by devshgraphicsprogramming** [microsoft/hlsl-specs#241]
**Which proposal does this relate to?**
[Please link the file in the repo where the problem is documented.](https://github.com/microsoft/hlsl-specs/blob/main/proposals/0011-inline-spirv.md)

**Describe the issue or outstanding question.**
Banging my head against the wall trying to make Combined Sampler Images, or PhysicalStoragePointers (BDA) using types that are not "wholly build from ground up with `vk::SpirvType`"

For example I have
```cpp
struct MyAggregate
{
   int32_t a,b;
};
```

For a regular `vk::RawBufferLoad` or a `StructuredBuffer`, DXC seems to make some meta-types that don't match `vk::SpirvType<..,T>` I'd make myself.

This gets in the way of doing `OpAccessChain` or similar things, because the types don't have the same `%result_id` from an `OpType..` even though the types may be identically defined.

**Additional context**
I've outlined list of vastly different currently open DXC issues that may touch upon this here:
https://github.com/microsoft/DirectXShaderCompiler/issues/6578#issuecomment-2103318352

I'd also settle for getting the `vk::spirvType` of a type instead of an expression, this would be enough to e.g. have the exact same type that a `Texture2D` is using, or getting the basic `MyAggregate` typedef to make pointers out of.

**Comment by s-perron** [comment]
I don't think we need anything new in HLSL to handle this. The issue is the compiler implementation. The errors can vary depending on the situation. Some times there are issues with the compiler duplicating types in the spir-v. We have a solution for that with pointers. This will be harder with structs.

Another problem is how to specify the layout. There is no good solution for that in DXC.

We will leave this open to remind us of this when implementing inline spir-v in clang.
